test-data/unit/check-protocols.test

```diff
@@ -2621,7 +2621,12 @@ a: Other
 b: Bad
 
 func(a)
-func(b)  # E: Argument 1 to "func" has incompatible type "Bad"; expected "One"
```

Comment:
how did this test already exist if mypy wasnt checking the argument names before?

Reply:
Because `__call__` was special-cased to check the arg names properly.